src/_gettsim/interface.py

```diff
@@ -373,6 +359,57 @@ def _create_input_data(
     return input_data
 
 
+def _warn_if_functions_and_columns_overlap(
+    columns_overriding_functions: set[str],
+) -> None:
+    """Warn if functions which compute columns overlap with existing columns.
+
+    Parameters
+    ----------
+    columns_overriding_functions : set[str]
+        Names of columns in the data that override hard-coded functions.
+    """
+    if columns_overriding_functions:
+        n_cols = len(columns_overriding_functions)
+        first_part = format_errors_and_warnings(
+            f"Your data provides the column{'' if n_cols == 1 else 's'}:"
+        )
+        formatted = format_list_linewise(list(columns_overriding_functions))
+        second_part = format_errors_and_warnings(
+            f"""
+            {'This is' if n_cols == 1 else 'These are'} already present among the
+            hard-coded functions of the taxes and transfers system.
+            If you want {'this' if n_cols == 1 else 'a'} data column to be used
+            instead of calculating it within GETTSIM you need not do anything.
+            If you want {'this' if n_cols == 1 else 'a'} data column to be calculated
+            by hard-coded functions, remove {'it' if n_cols == 1 else 'them'} from the
+            *data* you pass to GETTSIM. {'' if n_cols == 1 else '''You need to pick one
+            option for each column that appears in the list above.'''}
+            """
+        )
+        how_to_ignore = format_errors_and_warnings(
+            """
+            If you want to ignore this warning, add the following code to your script
+            before calling GETTSIM:
+
+                import warnings
+                warnings.filterwarnings(
+                    "ignore",
+                    category=FunctionsAndColumnsOverlapWarning
+                )
+            """
+        )
+        warnings.warn(
+            f"{first_part}\n{formatted}\n{second_part}\n{how_to_ignore}",
+            category=FunctionsAndColumnsOverlapWarning,
+            stacklevel=2,
+        )
+
+
+class FunctionsAndColumnsOverlapWarning(UserWarning):
```

Comment:
Apologies for being pedantic, but can't we include the string with the warning in the class itself now? I would find that much more natural.

Actually, we might not need the function `_warn_if_functions_and_columns_overlap` at all anymore. That is, the calling code could be (trying the walrus without code checking here, very adventurous!):

```py
if columns_overriding_functions := set(functions_overridden):
    warnings.warn(
        FunctionsAndColumnsOverlapWarning(columns_overriding_functions),
        stacklevel=2,
    )
```

and the warning class something like:


```py
class FunctionsAndColumnsOverlapWarning(UserWarning):
    def __init__(self, columns_overriding_functions):
        super().__init__(self.message(columns_overriding_functions))

    def message(self, columns_overriding_functions):
        return f"(repeat previous): {columns_overriding_functions}"
```